internal/util/variables.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2024 Humanitec
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import "regexp"
+
+// PrepareEnvVariables replaces the dollar sign ($) by double dollar sign ($$)
+// in slice strings for using in the docker compose file
+func PrepareEnvVariables(arr []string) []string {
+	// prepare replace string
+	re := regexp.MustCompile(`([\w|\s|\-|\}|\)])\$([\w+|\{?])`)
```

Comment:
This seems to cover most cases except for a string that starts with `$` like `"$VAR"`. Could you fix that and add a test please?

This may happen when someone has `command: ["/some/binary"], args: ["$MY", "${VARIABLE}"]` or something like that.

Reply:
Oh yes, you're right. I overlooked that. I'll fix that.